templater/templater.go

```diff
@@ -7,20 +7,42 @@ import (
 	"github.com/opencontrol/fedramp-templater/ssp"
 )
 
-// TemplatizeSSP inserts OpenControl data into (i.e. modifies) the provided SSP.
-func TemplatizeSSP(s *ssp.Document, openControlData opencontrols.Data) (err error) {
+func fillSummaryTables(s *ssp.Document, openControlData opencontrols.Data) (err error) {
 	tables, err := s.SummaryTables()
 	if err != nil {
 		return
 	}
 	for _, table := range tables {
-		ct := control.SummaryTable{Root: table}
+		ct := control.NewSummaryTable(table)
```

Comment:
Now that we are getting a common way to do `Fill` and `Diff` and we have these exported constructors `NewSummaryTable` and `NewNarrativeTable` we can define a common interface and have those constructors return that interface.

```go
type Actionable interface { //the name is horrendous. we can change later.i'm really bad at names
     Fill(opencontrols.Data) error
     Diff(opencontrols.Data) error
}
```

then return it instead of the struct.

```go
func NewNarrativeTable(root xml.Node) Actionable {
```
then we can unexport / lowercase SummaryTable and NarrativeTable.

this interface would be in it's own package.